flutter_local_notifications/lib/src/platform_specifics/android/schedule_mode.dart

```diff
@@ -3,6 +3,11 @@
 /// This leverages the use of alarms to schedule notifications as described
 /// at https://developer.android.com/training/scheduling/alarms
 enum AndroidScheduleMode {
+  /// Used to specify that the notification should be scheduled to be shown at
```

Comment:
This could be from lack of knowledge with the API but given the description, what is actually different between using this and `exactAlarmWhileIdle` then? Looking at the docs https://developer.android.com/reference/androidx/core/app/AlarmManagerCompat#setAlarmClock(android.app.AlarmManager,long,android.app.PendingIntent,android.app.PendingIntent), it says it would be same as `setExact` when using the `AlarmManagerCompat` APIs with `RTC_WAKEUP`. `RTC_WAKEUP` is already used as a flag in the exact scenario.

Reply:
I actually tried all different versions of setting exact alarm and nothing worked for me to deliver notification and ignore the low power mode.
For example if you are making an alarm app, it doesn't make sense to fire the alarm after half an hour or ignore it when power is low. So I agree with you that the difference is RTC_WAKEUP but for APIs older than 21. This is the code of this method:

```java
    @SuppressLint("MissingPermission")
    public static void setAlarmClock(@NonNull AlarmManager alarmManager, long triggerTime,
            @NonNull PendingIntent showIntent, @NonNull PendingIntent operation) {
        if (Build.VERSION.SDK_INT >= 21) {
            Api21Impl.setAlarmClock(alarmManager,
                    Api21Impl.createAlarmClockInfo(triggerTime, showIntent), operation);
        } else {
            AlarmManagerCompat.setExact(alarmManager, RTC_WAKEUP, triggerTime, operation);
        }
    }
```

This requires SCHEDULE_EXACT_ALARM too like setExact. But this will show alarm icon on the status bar when you use it like this in the screenshot.

![mmEq9](https://github.com/MaikuB/flutter_local_notifications/assets/735154/75b4d5d6-0033-4a77-8eac-186e926977de)


If you try to set an alarm on your device now, you will see that this icon will show up. Disabling the alarm will let it disappear only if there is no other app sets alarms.

Reply:
@MaikuB  I added an example under a button called user alarm clock to schedule...

Reply:
Thanks for the explanation